src/components/common/App/AppInput.vue

```diff
@@ -9,28 +12,124 @@ export default defineComponent({
     modelValue: {
       type: String,
       required: true,
+    },
+    label: {
+      type: String,
+    },
+    type: {
+      type: String,
+      default: 'text'
+    },
+    isValid: {
+      type: Boolean,
+      default: true,
+    },
+  },
+  data() {
+    return {
+      passwordChangingType: 'password',
+      eyeClicked: false,
+    }
+  },
+  emits: ['update:modelValue', 'blur'],
+  computed: {
+    inputType() {
+      return this.type === 'password' ? this.passwordChangingType : this.type
     }
   },
-  emits: ['update:modelValue'],
+  methods: {
+    toggleVisibility() {
+      this.passwordChangingType === 'password'
+        ? this.passwordChangingType = 'text'
+        : this.passwordChangingType = 'password'
+      this.eyeClicked = !this.eyeClicked
+    }
+  }
 });
 </script>
+
 <template>
-  <input class="input"
-    :placeholder="placeholder"
-    :value="modelValue"
-    @input="$emit('update:modelValue', $event.target.value)"
-  />
+  <div class="app-input">
+    <AppLabel>{{ label }}</AppLabel>
+    <input class="app-input__field"
+      :type="inputType"
+      :placeholder="placeholder"
+      :value="modelValue"
+      :isValid="isValid"
+      @input="$emit('update:modelValue', $event.target.value)"
+      @blur="$emit('blur', $event)"
+    />
+    <button
+      v-if="type === 'password'"
+      type="button"
+      class="app-input__visibility-toggle"
+      @click="toggleVisibility()"
+    >
+      <img
+        v-if="passwordChangingType === 'password'"
+        class="app-input__visibility-toggle--icon"
+        src='@/assets/PasswordEye.svg'
+        :clicked="!eyeClicked"
+      />
+      <img
+        v-if="passwordChangingType === 'text'"
+        class="app-input__visibility-toggle--icon"
+        src='@/assets/PasswordEyeOff.svg'
+        :clicked="eyeClicked"
+      />
+    </button>
```

Comment:
So, one thing to keep in mind, especially going into the future - is that while all of the above works, our AppInput is getting a liiittle big - it does a lot of things, and conditionally - even more.
We should actually use this base component - because it is in heart a base input for all the inputs in the app as it is intended and create a custom component just for the password handling - say `PasswordInput`.
It could use some slots to pass a button with the icons (which would trim out the code out of the `AppInput`) and would handle only the password input type and the toggle (and pass any other appropriate props to the `AppInput`).
This isn't something  I'd do in *this* PR (as it's veeeery big already) but definitely something to have in mind for the future.